### ISSUE TITLE
Blurry?

### ISSUE THREAD

**Issue by thesombady** [itsjunetime/tdf#13]
Hi, I just saw this and wanted to give it a go! However, once installing it and making a quick alias, the rendering is really buggy.
![Screenshot 2024-09-08 at 11 28 08](https://github.com/user-attachments/assets/636226e8-8130-4ff6-beb0-fa0e3dc3447c)

Have I installed it wrong?
I just followed the steps in the readme.

Hope I can get a fix, because this would be awesome otherwise!

**Comment by thesombady** [comment]
A quick note, I use WARP as my terminal emulator, and I tried using iterm2 and it worked, but iterm2 is really not a good emulator anymore.

**Comment by alerque** [comment]
I believe this is a function of your terminal not supporting the image protocols needed for high res images and it is falling back to a Unicode block drawing algorithm. Some terminals support image drawing protocols and some don't. There are two or three major ones I think, sixel, kitty, and iterm. Some of the terminals I have available on Linux support sixel or kitty and those seem to work fine with `tdf`, the ones that don't support it look low resolution like you show.

In spite of your attempt to say iterm2 is not a good terminal and WARP is, I think you need to take this up with WARP which doesn't seem to support any image protocols. Or if they do, maybe report which ones they do and then `tdf` can figure out how to detect/categorize it.

**Comment by alerque** [comment]
https://github.com/warpdotdev/Warp/issues/26

**Comment by thesombady** [comment]
Thanks! My argument for why warp is a good emulator is rooted in the fact that you can have panes without tmux, unlike many others. But I've sent to Warp and thanks for answering!

**Comment by alerque** [comment]
I think the [issue you opened](https://github.com/warpdotdev/Warp/issues/5286) is going to be a duplicate of the [one I already linked](https://github.com/warpdotdev/Warp/issues/26) but we'll leave that up to them to decide.

In any case I'll also note using `tmux` will probably block this, I don't think it supports pass through of any image protocols, but I could be wrong about that. At least it doesn't work in Kitty in a quick test I just did. [`zellij`](https://zellij.dev/) might work too but it also didn't work in a quick test (I don't use it regularly so I don't know if there is something to setup or enable).

For panes I use a tiling window manager anyway so it doesn't really matter if my terminal emulator supports it. My editor does, my terminal multiplexer does, and the window manager does, so I don't need another level of splits.
